unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Animator/AnimatorRegister.cs

```diff
@@ -11,7 +13,17 @@ public class AnimatorRegister : IDisposable
 
         public AnimatorRegister(int componentId, ECSComponentsFactory factory, IECSComponentWriter componentWriter)
         {
-            factory.AddOrReplaceComponent(componentId, AnimatorSerializer.Deserialize, () => new AnimatorComponentHandler(DataStore.i.ecs7));
+            var poolWrapper = new ECSReferenceTypeIecsComponentPool<PBAnimator>(
+                new WrappedComponentPool<IWrappedComponent<PBAnimator>>(10,
+                    () => new ProtobufWrappedComponent<PBAnimator>(new PBAnimator()))
```

Comment:
It can be further generalized as all PB Components are created by a default constructor. After all, if you don't require to specify default capacity pool creation can be a generic static one-liner

Reply:
PB Components have too many divergencies. I've tried to come up with a solution, but I would end up creating a static "ComponentPoolFactory" that would just shorten the Register method by one line.
Besides that, the pool capacities will be fine-tweaked in the future, because at the moment we don't know what would be the optimal amount for each kind of component.